### PR TITLE
Allow specifying ipv6 addresses

### PR DESCRIPTION
**Pull request by alimg**
If your ISP supports ipv6 like mine, chances are you are already getting a persistent address already without having to pay for a static ipv4 address. This commit enables us to send magic packets to ipv6 targets.

### CODE REVIEW

File: wakeonlan.py [hunk 1]
```diff
@@ -39,6 +40,7 @@ def send_magic_packet(
     ip_address: str = BROADCAST_IP,
     port: int = DEFAULT_PORT,
     interface: Optional[str] = None
+    address_family: socket.AddressFamily = socket.AF_INET
```
Comment (remcohaszing) [inline wakeonlan.py:43]: Could you add a docstring for this?

File: wakeonlan.py [hunk 2]
```diff
@@ -56,7 +58,13 @@ def send_magic_packet(
     """
     packets = [create_magic_packet(mac) for mac in macs]
 
-    with socket.socket(socket.AF_INET, socket.SOCK_DGRAM) as sock:
+    try:
+        if isinstance(ipaddress.ip_address(ip_address), ipaddress.IPv6Address):
+            address_family = socket.AF_INET6
+    except ValueError:
+        pass
```
Comment (remcohaszing) [inline wakeonlan.py:65]: Could you use `contextlib.ignored()` instead of a try/catch?

File: wakeonlan.py [hunk 3]
```diff
@@ -100,7 +114,10 @@ def main(argv: Optional[List[str]] = None) -> None:
         help="The ip address of the network adapter to route the magic packet through.",
     )
     args = parser.parse_args(argv)
-    send_magic_packet(*args.macs, ip_address=args.i, port=args.p, interface=args.n)
+    send_magic_packet(
+        *args.macs, 
+        ip_address=args.i, port=args.p, interface=args.n, 
+        address_family=socket.AF_INET6 if args.use_ipv6 else None)
```
Comment (remcohaszing) [inline wakeonlan.py:120]:
```suggestion
        address_family=socket.AF_INET6 if args.use_ipv6 else socket.AF_INET6)
```

**Review by remcohaszing — CHANGES_REQUESTED** [review]
Thank you!

Looks mostly good. Also I fixed CI upstream. Could you rebase and process the suggestions?

**Comment by infalmo** [comment]
Hi, can this be merged? Would appreciate ipv6 support, thanks!

**Comment by remcohaszing** [comment]
Sure! Could you rebase this, process the feedback, and provide a new PR?

Ideally could you also verify this works? I’m inclined to believe it does, but I can’t test it myself.

File: wakeonlan.py (hunk 2 above)
Comment (alimg) [reply wakeonlan.py]: I think you meant `contextlib.suppressed()` here, but I had to change this bit so I don't think you'd still want that in the new revision. No overstepping the user supplied value if the caller chooses to provide an explicit address_family now.

**Comment by alimg** [comment]
Sorry for long radio silence, I forgot I had this PR for a while. I address the comments and cleaned up a bit further. Hope you liked it better now.

Unfortunately my ISP decided not to give me an IPv6 address these days. Any help is appreciated to test if this works.

File: wakeonlan.py (hunk 2 above)
Comment (remcohaszing) [reply wakeonlan.py]: Yep. That’s what I meant, but I think your new changes are even better.

**Comment by remcohaszing** [comment]
Could you fix CI? I tried, but I’m not allowed to push to your fork.

**Comment by remcohaszing** [comment]
Released in version [3.1.0](https://github.com/remcohaszing/pywakeonlan/releases/tag/3.1.0) :tada:

**Pull request merged**